docs/guides/javascript/reactive/index.md

```diff
@@ -0,0 +1,1382 @@
+
+The method will receive an object called `descriptor` which contains the data structure passed to the constructor. This method can define its own CSS selectors, classes, or any other constant needed, so defining it as the first method in the class is a good practice.
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+The typical create method looks like:
+
+```js
+create(descriptor) {
+    // An optional component name for debugging.
+    this.name = 'courseindex_cm';
+
+    // Those are the component CSS selectors, in uppercase
+    // are used as constants.
+    this.selectors = {
+        EXAMPLESELECTOR: `[data-for='example']`,
+    };
+
+    // If the component manage CSS classes, the constants should
+    // be defined the same way as the selectors.
+    this.classes = {
+        HIDDEN: 'dimmed',
+    };
+
+    // The main HTML element is stored in "this.element" as well
+    // in "descriptor.element".
+    this.id = this.element.dataset.id;
+
+    (... any other data extracted from the "descriptor"...)
+}
+```
+
+  </div>
+</details>
+
+:::note
+
+When create is called the component has an `element` attribute because it should read data from the main element but it does not have a `this.reactive` attribute because the component can be registered before having a fully functional Reactive Instance. Any initialization that requires a reactive instance or a state should be done in later hooks like stateReady or getWatchers.
+
+:::
+
+  </TabItem>
+  <TabItem value="getWatchers" label="getWatchers">
+
+At the moment the Reactive Instance is registering the component and asking it what parts of the state it is watching. This method returns a list of state changes the component is watching and the methods that will handle those changes, something like an event listener for state changes.
+
+#### The `getWatchers(): array` hook
+
+This hook is called when the Reactive Instance is registering the component. At this moment the component can specify which state changes are watching and what methods will use to listen to those changes.
+
+The method should return an array of objects containing:
+watch: the state changes to watch. For example `user.username:update`
+handler: the class method to handle the state change event. For example `this._refreshUsername`.
+
+<details>
+  <summary>View examples</summary>
+  <div>
+
+This is a real example of getWatchers from a course index module:
+
+```js
+getWatchers() {
+    return [
+        {watch: `cm:created`, handler: this._createCm},
+        {watch: `cm:deleted`, handler: this._deleteCm},
+        {watch: `section:created`, handler: this._createSection},
+        {watch: `section:deleted`, handler: this._deleteSection},
+    ];
+}
+```
+
+In case the component only watches changes in a specific element of a state collection, the function could be something like.
+
+```js
+getWatchers() {
+    const id = this.element.dataset.id;
```

Comment:
It may be worth mentioning where the `this.element` is specified elsewhere in the code and how it comes to be here.

Reply:
done.